File: coldstart-mcp/tests/fixtures/python/token_service.py
"""Token service."""
import base64
import json
from typing import Optional


class TokenService:
    def __init__(self, secret: str):
        self.secret = secret

    def sign(self, payload: dict) -> str:
        return base64.b64encode(json.dumps(payload).encode()).decode()

    def verify(self, token: str) -> Optional[str]:
        try:
            payload = json.loads(base64.b64decode(token).decode())
            return payload.get('user_id')
        except Exception:
            return None

File: tests/fixtures/typescript/tokenService.ts
export class TokenService {
  constructor(private readonly secret: string) {}

  sign(payload: Record<string, unknown>): string {
    // Simplified — real implementation uses jsonwebtoken
    return Buffer.from(JSON.stringify(payload)).toString('base64');
  }

  verify(token: string): string {
    const payload = JSON.parse(Buffer.from(token, 'base64').toString('utf-8'));
    return payload.userId as string;
  }
}

export const defaultTokenService = new TokenService(process.env['JWT_SECRET'] ?? 'dev');
